code/typescript/pipelines-workshop/lib/pipeline-stack.ts

```diff
@@ -14,55 +12,49 @@ export class WorkshopPipelineStack extends cdk.Stack {
             repositoryName: "WorkshopRepo"
         });
 
-        // Defines the artifact representing the sourcecode
-        const sourceArtifact = new codepipeline.Artifact();
-        // Defines the artifact representing the cloud assembly
-        // (cloudformation template + all other assets)
-        const cloudAssemblyArtifact = new codepipeline.Artifact();
-
         // The basic pipeline declaration. This sets the initial structure
         // of our pipeline
-        const pipeline = new CdkPipeline(this, 'Pipeline', {
+
+        const pipeline = new CodePipeline(this, 'Pipeline', {
             pipelineName: 'WorkshopPipeline',
-            cloudAssemblyArtifact,
+            synth: new CodeBuildStep('SynthStep', {
+                    input: CodePipelineSource.codeCommit(repo, 'master'),
+                    installCommands: [
+                        'npm install -g aws-cdk'
+                    ],
+                    commands: [
+                        'npm ci',
+                        'npm run build',
+                        'npx cdk synth'
+                    ]
+                }
+            )
+        });
 
-            // Generates the source artifact from the repo we created in the last step
-            sourceAction: new codepipeline_actions.CodeCommitSourceAction({
-                actionName: 'CodeCommit', // Any Git-based source control
-                output: sourceArtifact, // Indicates where the artifact is stored
-                repository: repo // Designates the repo to draw code from
+        const deploy = new WorkshopPipelineStage(this, 'Deploy');
+        const deployStage = pipeline.addStage(deploy);
+
+        deployStage.addPost(
+            new CodeBuildStep('TestViewerEndpoint', {
+                projectName: 'TestViewerEndpoint',
+                envFromCfnOutputs: {
+                    ENDPOINT_URL: deploy.hcViewerUrl
+                },
+                commands: [
+                    'curl -Ssf $ENDPOINT_URL'
+                ]
             }),
-
-            // Builds our source code outlined above into a could assembly artifact
-            synthAction: SimpleSynthAction.standardNpmSynth({
-                sourceArtifact, // Where to get source code to build
-                cloudAssemblyArtifact, // Where to place built source
-
-                buildCommand: 'npm run build' // Language-specific build cmd
+            new CodeBuildStep('TestAPIGatewayEndpoint', {
+                projectName: 'TestAPIGatewayEndpoint',
+                envFromCfnOutputs: {
+                    ENDPOINT_URL: deploy.hcEndpoint
+                },
+                commands: [
+                    'curl -Ssf $ENDPOINT_URL',
+                    'curl -Ssf $ENDPOINT_URL',
```

Comment:
This is supposed to be different sub-urls I think

```suggestion
                    'curl -Ssf $ENDPOINT_URL/hello',
```